library/src/main/java/io/github/luizgrp/sectionedrecyclerviewadapter/SectionedRecyclerViewAdapter.java

```diff
@@ -86,68 +86,104 @@ public SectionedRecyclerViewAdapter() {
     }
 
     private RecyclerView.ViewHolder getItemViewHolder(ViewGroup parent, Section section) {
-        View view = LayoutInflater.from(parent.getContext()).inflate(section.getItemResourceId(), parent, false);
-        // get the item viewholder from the section
+        View view;
+        if (section.isItemViewWillBeProvided()) {
+            view = section.getItemView(parent);
+            if (view == null) {
+                throw new NullPointerException("Section.getItemView() returned null");
+            }
+        } else {
+            Integer resId = section.getItemResourceId();
+            if (resId == null) {
+                throw new NullPointerException("Missing 'item' resource id");
+            }
+            view = LayoutInflater.from(parent.getContext()).inflate(section.getItemResourceId(), parent, false);
+        }
```

Comment:
are you up for the "challenge" of adding unit tests for these getViewHolders methods? if so I can open an issue with the design and it can be done in a separate PR after this one gets merged

Reply:
Wouldn't those be the same tests as the ones you suggested above? I.e. ensuring that SectionedRecyclerViewAdapter.onCreateViewHolder calls through the right Section.getXyzView() and/or Section.getXyzViewHolder() function?

Reply:
Sorry if I suggested the same thing. Yes, it should test as you described.
We will have to extract this LayoutInflater call to a class that can be mocked in the tests so we don't need to add powermock or robolectric.

Reply:
Done. Moved to package-private inflate() method that can be overridden in the unit test.